app/controllers/lti_nrps_controller.rb

```diff
@@ -106,44 +122,37 @@ def parse_members_data(lcd, members_data)
 
       next if user_data.key?("status") && user_data["status"] != "Active"
 
-      cud_data = {}
+      next unless user_data.key?("email") && user_data.key?("given_name") &&
+                  user_data.key?("family_name")
+
       # Normalize email
       user_data["email"].downcase!
 
+      cud_data = {}
       user = User.find_by(email: user_data["email"])
-      cud_data[:color] = if user.nil? || lcd.course.course_user_data.find_by(user_id: user.id).nil?
-                           "green"
-                         else
-                           "black"
-                         end
-      cud_data[:email] = user_data["email"]
-      cud_data[:first_name] = user_data["given_name"]
-      cud_data[:last_name] = user_data["family_name"]
-      cud_view << cud_data
+      if user.nil? || lcd.course.course_user_data.find_by(user_id: user.id).nil?
+        cud_data[:color] = "green"
+        cud_data[:email] = user_data["email"]
+        cud_data[:first_name] = user_data["given_name"]
+        cud_data[:last_name] = user_data["family_name"]
+      else
+        cud = email_to_cud[user.email]
+        cud_data = populate_cud_data(cud)
+        cud_data[:color] = "black"
+      end
 
+      cud_view << cud_data
       email_to_cud.delete(cud_data[:email])
     end
 
     return cud_view unless lcd.drop_missing_students
 
     # Mark the remaining students as dropped
-    email_to_cud.each do |email, cud|
+    email_to_cud.each do |_email, cud|
```

Comment:
Nit: If `_email` is not needed, it is cleaner to use `_` as the variable name

Reply:
Will change